src/WebJobs.Script.WebHost/Diagnostics/DeferredLogger.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Channels;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    /// <summary>
+    /// A logger that defers log entries to a channel.
+    /// </summary>
+    public class DeferredLogger : ILogger
+    {
+        private readonly Channel<DeferredLogEntry> _channel;
+        private readonly string _categoryName;
+        private readonly List<string> _scopes = new List<string>();
+
+        public DeferredLogger(Channel<DeferredLogEntry> channel, string categoryName)
+        {
+            _channel = channel;
+            _categoryName = categoryName;
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
```

Comment:
Scope can be simplified via the usage of `ISupportExternalScope`.

Implement `ISupportExternalScope` on the `ILoggerProvider`. It will give you an `IExternalScopeProvider` - pass that to this logger and use it for `BeginScope` implementation.

Reply:
Added ISupportExternalScope